src/Carousel/widget/components/ImageCarousel.tsx

```diff
@@ -0,0 +1,274 @@
+
+import * as React from "Carousel/lib/react";
+
+import ImageCarouselModelProps, { HeightUnits, OnClickEvent,
+    PageLocation, WidthUnits } from "./../../Carousel.d";
+import { Data } from "./../Carousel";
+
+import Carousel, { CarouselProps } from "./Carousel";
+import CarouselCaption from "./CarouselCaption";
+import { CarouselItem, CarouselItemProps } from "./CarouselItem";
+
+interface ShowPageProps {
+    pageName?: string;
+    onClickEvent?: OnClickEvent;
+    pageLocation?: PageLocation;
+    context?: string;
+    guid?: string;
+}
+
+interface OnClickProps {
+    guid?: string;
+    page?: string;
+    microflow?: string;
+    pageLocation?: PageLocation;
+    onClickEvent?: OnClickEvent;
+}
+
+interface ItemProps extends CarouselItemProps {
+    key: string | number;
+    imgStyle: Object;
+    alt: string;
+    src: string;
+    caption: string;
+    description: string;
+}
+
+/**
+ * All properties of Image Carousel, 
+ * Including all props from the modeler, and props from react.
```

Comment:
If you need comments to describe what your interface is for, you're doing something wrong. The name of the interface should be clear and self-documenting.